WinUi3Template1/MainWindow.xaml.cs

```diff
@@ -12,29 +18,28 @@ public MainWindow()
             Title = Settings.FeatureName;
 
             var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
-            LoadIcon(hwnd,"Assets/windows-sdk.ico");
+            LoadIcon(hwnd, "Assets/windows-sdk.ico");
             SetWindowSize(hwnd, 1920, 1080);
         }
 
         private void LoadIcon(IntPtr hwnd, string iconName)
         {
+            var hIcon = LoadImage(null, iconName, GDI_IMAGE_TYPE.IMAGE_ICON, 16, 16, IMAGE_FLAGS.LR_LOADFROMFILE);
 
-            IntPtr hIcon = PInvoke.User32.LoadImage(IntPtr.Zero, iconName,
-                PInvoke.User32.ImageType.IMAGE_ICON, 16, 16, PInvoke.User32.LoadImageFlags.LR_LOADFROMFILE);
-
-            PInvoke.User32.SendMessage(hwnd, PInvoke.User32.WindowMessage.WM_SETICON, (IntPtr)0, hIcon);
+            var success = false;
+            hIcon.DangerousAddRef(ref success);
```

Comment:
I don't know why you'd need `DangerousAddRef` here. It doesn't appear that you're using it as described in [its doc](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.dangerousaddref?view=net-5.0#remarks) either. 
Maybe drop the `DangerousAddRef` and `DangerousRelease` method calls and add `using` in front of the `hIcon` local declaration so that the `SafeHandle` is disposed of when you're done with it automatically. The "before" code had a leak where it never released the handle, I guess. Unless the receiver of SendMessage does this. But in that case, I'm not exactly sure how to get a `SafeHandle` to *not* dispose the handle when it is finalized.

Reply:
When I add using I get: Exception thrown at 0x00007FF99F6872FA (ntdll.dll) in WinUi3Template1.exe: 0xC0000008: An invalid handle was specified.

Reply:
Hmm... the return type being `SafeFileHandle` is broken. The release method that the handle should be calling is `DestroyIcon` in this case. But per [the docs](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadimagea#return-value), any of three release methods may be appropriate based on the `type` parameter. I suspect that's too complex for the metadata to describe today. I filed https://github.com/microsoft/win32metadata/issues/611 to track.

The exception that's thrown to you now will be thrown from the finalizer and crash your app later if you don't dispose of it now. You should use the other overload that returns the handle instead of the safehandle for now, and call `DestroyIcon` directly as a workaround.